Starcraft2Client.py

```diff
@@ -111,6 +167,17 @@ def on_package(self, cmd: str, args: dict):
             for mission in slot_req_table:
                 self.mission_req_table[mission] = MissionInfo(**slot_req_table[mission])
 
+            # Look for and set SC2PATH; warn user if not found
+            try:
+                os.environ["SC2PATH"]
+            except KeyError:
```

Comment:
how about `if "SC2PATH" not in os.environ:` ?
or maybe even
```python
if "SC2PATH" not in os.environ and check_game_install_path():
    check_mod_install()
```
also we don't want to check_mod_install if SC2PATH is set? This is correct?

Reply:
Great suggestion. Self-education has its blind spots, and figuring out if a key is in a dict was one of mine.

My logic for not calling check_mod_install if SC2PATH was set prior to this snippet was as follows:
1) I figured the only way SC2PATH already exists is if the user calls /set_path manually, which includes a call to check_mod_install, so I didn't want to needlessly run it again.
2) The reason /set_path calls check_mod_install is because hypothetically, a user could fail this automatic check you've highlighted, then manually set the path. I wouldn't want a user to miss out on check_mod_install in that case.

I suppose this does miss the people who set an SC2PATH environment variable within their actual OS. 🤔 I find it hard to believe someone would do that, though, considering I don't think that's a thing outside of the sc2 package. I'll follow whatever suggestion you have for that.